UniqueException

fogelito opened the pull request:
Added support for UniqueException to differentiate it from DuplicateException.
UniqueException extends DuplicateException for backwards compatibility. It will be thrown for all unique constraint violations except those caused by the _uid constraint.

<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.

You can learn more about contributing to appwrite here: https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md

Happy contributing!

-->

## What does this PR do?

(Provide a description of what this PR does and why it's needed.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Screenshots may also be helpful.)

## Related PRs and Issues

- (Related PR or issue)

## Checklist

- [ ] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [ ] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

fogelito committed:
Run tests

fogelito committed:
Merge branch '1.8.x' of https://github.com/appwrite/appwrite into unique-exception

# Conflicts:
#	composer.lock

fogelito committed:
Merge branch '1.8.x' of https://github.com/appwrite/appwrite into unique-exception

fogelito committed:
linter

fogelito committed:
Add DOCUMENT_UNIQUE_VIOLATION

fogelito committed:
Remove var_dumps

fogelito committed:
linter

fogelito committed:
Remove var_dump

abnegate commented:
@copilot Sync with 1.9.x

Copilot committed:
Merge remote-tracking branch 'origin/1.9.x' into unique-exception

# Conflicts:
#	composer.json
#	composer.lock
#	tests/e2e/Services/Databases/DatabasesBase.php

Co-authored-by: abnegate <5857008+abnegate@users.noreply.github.com>

abnegate commented:
@copilot Update to database 5.* and run composer update. Completion means all CI checks will pass

Copilot committed:
chore: update utopia-php/database to stable 5.* and run composer update

Agent-Logs-Url: https://github.com/appwrite/appwrite/sessions/045dbae8-cb3c-42ae-bfa0-144d0680ba26

Co-authored-by: abnegate <5857008+abnegate@users.noreply.github.com>
